linked_list/linked_list.py

```diff
@@ -0,0 +1,246 @@
+    def remove_front(self):
+        # Deletes First Element
+        self.remove(0)
+
+    def remove_last(self):
+        # Deletes Last Element
+        self.remove(self.size - 1)
+
+    def remove(self, index):
+        # Deletes Element present at index
+        if self.is_empty():
+            raise ValueError
+        if index < 0 or index >= self.size:
+            raise IndexError
+        if index >= 0:
+            temp = self.head
+            if temp.next is None:
+                removed = temp.data
+                self.head = None
+            else:
+                for _ in range(index - 1):
+                    temp = temp.next
+                removed = temp.next.data
+                temp.next = temp.next.next
+            self.size -= 1
+            return removed
+
+    def search(self, inp):
+        # Search for inp in Linked List and return it's index
+        head = self.head
+        for i in range(self.size):
+            if head.data is inp:
+                return i
+            else:
+                head = head.next
+        raise ValueError
+
+    def find_first(self, inp):
+        # Finds First occurence of data in Linked List
+        return self.search(inp)
+
+    def find_last(self, inp):
+        # Finds Last occurence of data in Linked List
+        head = self.head
+        found = False
+        for i in range(self.size):
+            if head.data is inp:
+                index = i
+                found = True
+            head = head.next
+        if found:
+            return index
+        else:
+            raise ValueError
+
+    def set(self, data, index):
+        # Sets the data to index and replaces previously existing data.
+        if index >= self.size:
+            add(data, index)
+        else:
+            head = self.head
+            for i in range(self.size):
+                if i is index:
+                    head.data = data
+                head = head.next
+
+    def clear(self):
+        # Removes all elements from Linked List
+        self.head = None
+        self.size = 0
+
+    def is_empty(self):
+        # Checks if Linked List is Empty.
+        return self.size is 0
+
+    def clone(self):
+        # Clones the Linked List
+        new = Linked_list()
+        temp = self.head
+        while(temp.next):
+            new.add_last(temp.data)
+            temp = temp.next
+        new.add_last(temp.data)
+        return new
+
+
+def main():
+    l_list = Linked_list()
+
+    l_list.add_last(5)
+    l_list.__str__()
+    l_list.__len__()
+
+    l_list.add_last(15)
+    l_list.__str__()
+    l_list.__len__()
+
+    # Add Element at Front
+    l_list.add_front(10)
+    l_list.__str__()
+    l_list.__len__()
+
+    # Add Element at Front
+    l_list.add_front(11)
+    l_list.__str__()
+    l_list.__len__()
+
+    # Add Element at Last
+    l_list.add_last(6)
+    l_list.__str__()
+    l_list.__len__()
+
+    # Add Element at Front
+    l_list.add_front(5)
+    l_list.__str__()
```

Comment:
Similar if `__len__` is returning the length integer, you can show it as `print len(l_list)`